Traceback (most recent call last):
  File "C:\Users\unerue\miniconda3\envs\vision\lib\site-packages\jupyter_cache\executors\utils.py", line 51, in single_nb_execution
    executenb(
  File "C:\Users\unerue\miniconda3\envs\vision\lib\site-packages\nbclient\client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\unerue\miniconda3\envs\vision\lib\site-packages\nbclient\util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "C:\Users\unerue\miniconda3\envs\vision\lib\site-packages\nbclient\util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "C:\Users\unerue\miniconda3\envs\vision\lib\asyncio\base_events.py", line 642, in run_until_complete
    return future.result()
  File "C:\Users\unerue\miniconda3\envs\vision\lib\site-packages\nbclient\client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "C:\Users\unerue\miniconda3\envs\vision\lib\site-packages\nbclient\client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\unerue\miniconda3\envs\vision\lib\site-packages\nbclient\client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from skimage.io import imread
from skimage.color import rgb2ycbcr, gray2rgb
image = imread('images/skin.png')
proc_image = np.reshape(rgb2ycbcr(image), (-1, 3))
skin_score = skin_gmm.score_samples(proc_image[...,1:])
not_skin_score = not_skin_gmm.score_samples(proc_image[...,1:])
result = skin_score > not_skin_score
result = result.reshape(image.shape[0], image.shape[1])
result = np.bitwise_and(gray2rgb(255*result.astype(np.uint8)), image)
plt.figure(figsize=(20,10))
plt.subplot(121), plt.imshow(image), plt.axis('off'), plt.title('Original', size=20)
plt.subplot(122), plt.imshow(result), plt.axis('off'), plt.title('Skin Detected and Segmented with GMM', size=20)
plt.show()
------------------

[1;31m---------------------------------------------------------------------------[0m
[1;31mFileNotFoundError[0m                         Traceback (most recent call last)
Input [1;32mIn [2][0m, in [0;36m<cell line: 3>[1;34m()[0m
[0;32m      1[0m [38;5;28;01mfrom[39;00m [38;5;21;01mskimage[39;00m[38;5;21;01m.[39;00m[38;5;21;01mio[39;00m [38;5;28;01mimport[39;00m imread
[0;32m      2[0m [38;5;28;01mfrom[39;00m [38;5;21;01mskimage[39;00m[38;5;21;01m.[39;00m[38;5;21;01mcolor[39;00m [38;5;28;01mimport[39;00m rgb2ycbcr, gray2rgb
[1;32m----> 3[0m image [38;5;241m=[39m [43mimread[49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43mimages/skin.png[39;49m[38;5;124;43m'[39;49m[43m)[49m
[0;32m      4[0m proc_image [38;5;241m=[39m np[38;5;241m.[39mreshape(rgb2ycbcr(image), ([38;5;241m-[39m[38;5;241m1[39m, [38;5;241m3[39m))
[0;32m      5[0m skin_score [38;5;241m=[39m skin_gmm[38;5;241m.[39mscore_samples(proc_image[[38;5;241m.[39m[38;5;241m.[39m[38;5;241m.[39m,[38;5;241m1[39m:])

File [1;32m~\miniconda3\envs\vision\lib\site-packages\skimage\io\_io.py:53[0m, in [0;36mimread[1;34m(fname, as_gray, plugin, **plugin_args)[0m
[0;32m     50[0m         plugin [38;5;241m=[39m [38;5;124m'[39m[38;5;124mtifffile[39m[38;5;124m'[39m
[0;32m     52[0m [38;5;28;01mwith[39;00m file_or_url_context(fname) [38;5;28;01mas[39;00m fname:
[1;32m---> 53[0m     img [38;5;241m=[39m call_plugin([38;5;124m'[39m[38;5;124mimread[39m[38;5;124m'[39m, fname, plugin[38;5;241m=[39mplugin, [38;5;241m*[39m[38;5;241m*[39mplugin_args)
[0;32m     55[0m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m [38;5;28mhasattr[39m(img, [38;5;124m'[39m[38;5;124mndim[39m[38;5;124m'[39m):
[0;32m     56[0m     [38;5;28;01mreturn[39;00m img

File [1;32m~\miniconda3\envs\vision\lib\site-packages\skimage\io\manage_plugins.py:207[0m, in [0;36mcall_plugin[1;34m(kind, *args, **kwargs)[0m
[0;32m    203[0m     [38;5;28;01mexcept[39;00m [38;5;167;01mIndexError[39;00m:
[0;32m    204[0m         [38;5;28;01mraise[39;00m [38;5;167;01mRuntimeError[39;00m([38;5;124m'[39m[38;5;124mCould not find the plugin [39m[38;5;124m"[39m[38;5;132;01m%s[39;00m[38;5;124m"[39m[38;5;124m for [39m[38;5;132;01m%s[39;00m[38;5;124m.[39m[38;5;124m'[39m [38;5;241m%[39m
[0;32m    205[0m                            (plugin, kind))
[1;32m--> 207[0m [38;5;28;01mreturn[39;00m func([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs)

File [1;32m~\miniconda3\envs\vision\lib\site-packages\skimage\io\_plugins\imageio_plugin.py:10[0m, in [0;36mimread[1;34m(*args, **kwargs)[0m
[0;32m      8[0m [38;5;129m@wraps[39m(imageio_imread)
[0;32m      9[0m [38;5;28;01mdef[39;00m [38;5;21mimread[39m([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs):
[1;32m---> 10[0m     [38;5;28;01mreturn[39;00m np[38;5;241m.[39masarray(imageio_imread([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs))

File [1;32m~\miniconda3\envs\vision\lib\site-packages\imageio\core\functions.py:265[0m, in [0;36mimread[1;34m(uri, format, **kwargs)[0m
[0;32m    260[0m     [38;5;28;01mraise[39;00m [38;5;167;01mTypeError[39;00m(
[0;32m    261[0m         [38;5;124m'[39m[38;5;124mInvalid keyword argument [39m[38;5;124m"[39m[38;5;124mmode[39m[38;5;124m"[39m[38;5;124m, [39m[38;5;124m'[39m [38;5;124m'[39m[38;5;124mperhaps you mean [39m[38;5;124m"[39m[38;5;124mpilmode[39m[38;5;124m"[39m[38;5;124m?[39m[38;5;124m'[39m
[0;32m    262[0m     )
[0;32m    264[0m [38;5;66;03m# Get reader and read first[39;00m
[1;32m--> 265[0m reader [38;5;241m=[39m read(uri, [38;5;28mformat[39m, [38;5;124m"[39m[38;5;124mi[39m[38;5;124m"[39m, [38;5;241m*[39m[38;5;241m*[39mkwargs)
[0;32m    266[0m [38;5;28;01mwith[39;00m reader:
[0;32m    267[0m     [38;5;28;01mreturn[39;00m reader[38;5;241m.[39mget_data([38;5;241m0[39m)

File [1;32m~\miniconda3\envs\vision\lib\site-packages\imageio\core\functions.py:172[0m, in [0;36mget_reader[1;34m(uri, format, mode, **kwargs)[0m
[0;32m    149[0m [38;5;124;03m""" get_reader(uri, format=None, mode='?', **kwargs)[39;00m
[0;32m    150[0m 
[0;32m    151[0m [38;5;124;03mReturns a :class:`.Reader` object which can be used to read data[39;00m
[1;32m   (...)[0m
[0;32m    168[0m [38;5;124;03m    to see what arguments are available for a particular format.[39;00m
[0;32m    169[0m [38;5;124;03m"""[39;00m
[0;32m    171[0m [38;5;66;03m# Create request object[39;00m
[1;32m--> 172[0m request [38;5;241m=[39m Request(uri, [38;5;124m"[39m[38;5;124mr[39m[38;5;124m"[39m [38;5;241m+[39m mode, [38;5;241m*[39m[38;5;241m*[39mkwargs)
[0;32m    174[0m [38;5;66;03m# Get format[39;00m
[0;32m    175[0m [38;5;28;01mif[39;00m [38;5;28mformat[39m [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m:

File [1;32m~\miniconda3\envs\vision\lib\site-packages\imageio\core\request.py:124[0m, in [0;36mRequest.__init__[1;34m(self, uri, mode, **kwargs)[0m
[0;32m    121[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124m'[39m[38;5;124mRequest requires mode[1] to be in [39m[38;5;124m"[39m[38;5;124miIvV?[39m[38;5;124m"[39m[38;5;124m'[39m)
[0;32m    123[0m [38;5;66;03m# Parse what was given[39;00m
[1;32m--> 124[0m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_parse_uri[49m[43m([49m[43muri[49m[43m)[49m
[0;32m    126[0m [38;5;66;03m# Set extension[39;00m
[0;32m    127[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39m_filename [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m:

File [1;32m~\miniconda3\envs\vision\lib\site-packages\imageio\core\request.py:260[0m, in [0;36mRequest._parse_uri[1;34m(self, uri)[0m
[0;32m    257[0m [38;5;28;01mif[39;00m is_read_request:
[0;32m    258[0m     [38;5;66;03m# Reading: check that the file exists (but is allowed a dir)[39;00m
[0;32m    259[0m     [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m os[38;5;241m.[39mpath[38;5;241m.[39mexists(fn):
[1;32m--> 260[0m         [38;5;28;01mraise[39;00m [38;5;167;01mFileNotFoundError[39;00m([38;5;124m"[39m[38;5;124mNo such file: [39m[38;5;124m'[39m[38;5;132;01m%s[39;00m[38;5;124m'[39m[38;5;124m"[39m [38;5;241m%[39m fn)
[0;32m    261[0m [38;5;28;01melse[39;00m:
[0;32m    262[0m     [38;5;66;03m# Writing: check that the directory to write to does exist[39;00m
[0;32m    263[0m     dn [38;5;241m=[39m os[38;5;241m.[39mpath[38;5;241m.[39mdirname(fn)

[1;31mFileNotFoundError[0m: No such file: 'C:\Users\unerue\Documents\unerue.github.io\computer-vision\images\skin.png'
FileNotFoundError: No such file: 'C:\Users\unerue\Documents\unerue.github.io\computer-vision\images\skin.png'

